helm/idol-nifi/templates/zookeeper/zookeeper.yaml

```diff
@@ -8,35 +8,38 @@
 # The information contained herein is subject to change without notice.
 #
 # END COPYRIGHT NOTICE
+{{- range $nifiClusterItem := .Values.nifiClusters }}
+{{ $nifiCluster := merge $nifiClusterItem $.Values.nifi (dict "clusterId" $.Values.name) }}
+---
 apiVersion: apps/v1
 kind: StatefulSet
 metadata:
-  name: zookeeper
-  labels: {{- include "idol-library.labels" (dict "root" . "component" .Values.zookeeper) | nindent 4 -}}
-    name: zookeeper
-    app: zookeeper
+  name: {{ $nifiCluster.clusterId }}-zk
+  labels: {{- include "idol-library.labels" (dict "root" $ "component" $.Values.zookeeper) | nindent 4 -}}
+    name: {{ $nifiCluster.clusterId }}-zk
+    app: {{ $nifiCluster.clusterId }}-zk
   annotations:
-    app.kubernetes.io/name: zookeeper
-    app.kubernetes.io/part-of: nifi
+    app.kubernetes.io/name: {{ $nifiCluster.clusterId }}-zk
+    app.kubernetes.io/part-of: {{ $nifiCluster.clusterId }}
 spec:
   revisionHistoryLimit: 1
-  serviceName: zookeeper
+  serviceName: {{ $nifiCluster.clusterId }}-zk
   podManagementPolicy: OrderedReady
   selector:
     matchLabels:
-      app: zookeeper
+      app: {{ $nifiCluster.clusterId }}-zk
   template:
     metadata:
-      labels: {{- include "idol-library.labels" (dict "root" . "component" .Values.zookeeper) | nindent 8 -}}
-        app: zookeeper
+      labels: {{- include "idol-library.labels" (dict "root" $ "component" $.Values.zookeeper) | nindent 8 -}}
+        app: {{ $nifiCluster.clusterId }}-zk
     spec:
       automountServiceAccountToken: false
       enableServiceLinks: false
       setHostnameAsFQDN: true
       dnsPolicy: ClusterFirstWithHostNet
       restartPolicy: Always
-{{- if .Values.podSecurityContext.enabled }}
-      securityContext: {{- omit .Values.podSecurityContext "enabled" | toYaml | nindent 8 }}
+{{- if $.Values.podSecurityContext.enabled }}
+      securityContext: {{- omit $.Values.podSecurityContext "enabled" | toYaml | nindent 8 }}
 {{- end }}
       imagePullSecrets:
       - name: dockerhub-secret
```

Comment:
Existing issue: this should source from `global.imagePullSecrets` rather than being hardcoded.